### PR TITLE
RCLL-41 Add endpoint to get all recalls for a prisoner

### PR DESCRIPTION
**Pull request by yasinmustafa**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/jpa/repository/RecallRepository.kt [hunk 1]
```diff
@@ -6,4 +6,5 @@ import java.util.UUID
 
 interface RecallRepository : CrudRepository<RecallEntity, Int> {
   fun findOneByRecallUniqueIdentifier(recallUniqueIdentifier: UUID): RecallEntity?
+  fun findByPrisonerId(prisonerId: String): List<RecallEntity>
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/jpa/repository/RecallRepository.kt:9]: Potentially may need to paginate this - just being used fir the POC at the moment so no decision yet

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/controller/RecallController.kt [hunk 2]
```diff
@@ -57,4 +57,19 @@ class RecallController(private val recallService: RecallService) {
     ],
   )
   fun getRecall(@PathVariable recallUuid: UUID): Recall = recallService.findRecallByUuid(recallUuid)
+
+  @GetMapping("/person/{prisonerId}")
+  @PreAuthorize("hasAnyRole('ROLE_REMAND_AND_SENTENCING', 'ROLE_RELEASE_DATES_CALCULATOR',  'ROLE_REMAND_SENTENCING__RECORD_RECALL_RW')")
```
Comment (ldlharper) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/controller/RecallController.kt:62]: Are these roles correct? I'd expect it to just be `ROLE_REMAND_SENTENCING__RECORD_RECALL_RO` and be creating a new read only role.

Comment (yasinmustafa) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/controller/RecallController.kt:62]: yes I think you're right - but I'd like to defer that please and discuss with @carloveo-moj and @joelstobart-moj 

will have to modify the system client to add the correct roles and then will come bck and refactor these endpoints

**Review by ldlharper — APPROVED** [review]
LGTM

**Pull request merged**